src/main/java/org/broadinstitute/hellbender/tools/spark/bwa/BwaSparkEngine.java

```diff
@@ -129,8 +133,17 @@ public void close() {
                 inputReads.add(readItr.next());
             }
             final int nReads = inputReads.size();
-            if (alignsPairs && (nReads & 1) != 0 ) {
-                throw new GATKException("We're supposed to be aligning paired reads, but there are an odd number of them.");
+            if ( alignsPairs ) {
+                if ( (nReads & 1) != 0 ) {
```

Comment:
Why are you using bitwise-AND instead of the mod operator to check whether the number is odd? Mod is more straightforward I think.

Reply:
Because Ted Sharpe.  Changed.